src/store.c

```diff
@@ -49,11 +52,13 @@ struct rill_packed header
     uint64_t keys;
     uint64_t pairs;
 
-    uint64_t vals_off;
     uint64_t data_off;
-    uint64_t index_off;
+    uint64_t data_b_off;
+
+    uint64_t index_off; // rename to index a
+    uint64_t index_b_off;
 
-    uint64_t reserved[3]; // for future use
+    uint64_t reserved[2]; // for future use
```

Comment:
Note that the value here was to keep the header to a cache-line and preserve alignment for the rest of the file. Since you added two u64 then I think reserved should be brought down to 1.

Note that this is mostly superficial and won't really affect performance. Also there should be a static assert or a comment for this as that requirement just only lives in my head.